src/plugins/backend_odbc/backend_odbc_set.c

```diff
@@ -0,0 +1,1277 @@
+ */
+
+#include "./backend_odbc_set.h"
+#include "./backend_odbc_general.h"
+#include "./backend_odbc_get.h"
+
+#include <kdbassert.h>
+#include <kdbdiff.h>
+#include <kdbease.h>
+#include <kdberrors.h>
+#include <kdbhelper.h>
+#include <kdblogger.h>
+#include <string.h>
+
+/* ODBC related includes */
+// #include <sql.h>
+#include <sqlext.h>
+
+
+/* Value for SQLLite, adjust for your ODBC driver/DBMS */
+#define ODBC_SQL_MAX_PARAMS 32766
+
+
+/**
+ * @internal
+ *
+ * @brief Get an SQL query for deleting metadata for ONE specific key and some metakeys
+ * 	If you want to delete ALL metakeys for one or more key-names, it's recommended to use the getDeleteQuery() function instead
+ *
+ * @param numMetaKeys The number of metakeys you want to remove (needed for number of parameters in IN(..) clause).
+ * @param dsConfig The data source configuration (contains mountpoint definition).
+ * @param quoteStr The char(s) that should be used for quoting identifiers.
+ * 	The identifiers in @dsConfig must not contain @p quoteString as a substring.
+ * @param[out] errorKey Used to store errors and warnings.
+ *
+ * @return The string with the query incl. parameter markers '?'. This query must be executed with a statement that has correctly bound
+ * parameters. Make sure to free the returned string.
+ * @retval NULL if an error occurred.
+ */
+static char * getDeleteMetaDataQuery (unsigned int numMetaKeys, const struct dataSourceConfig * dsConfig, const char * quoteStr,
+				      Key * errorKey)
+{
+	if (!dsConfig)
+	{
+		ELEKTRA_SET_INTERFACE_ERROR (errorKey, "The provided dataSourceConfig struct must not be NULL.");
+		return NULL;
+	}
+
+	if (!quoteStr || !(*quoteStr))
+	{
+		ELEKTRA_SET_INTERFACE_ERROR (errorKey, "NULL or empty strings are not supported for the quote string.");
+		return NULL;
+	}
+
+	if (numMetaKeys == 0)
+	{
+		ELEKTRA_ADD_INTERFACE_WARNING (errorKey,
+					       "The provided number of metakeys to delete was 0. "
+					       "Therefore, no query was created.");
+		return NULL;
+	}
+
+	/* Check if any of the used identifiers contains the quote string */
+	if (strstr (dsConfig->metaTableName, quoteStr))
+	{
+		ELEKTRA_SET_INTERFACE_ERRORF (errorKey, "The meta table name contained the quote string '%s', this is not allowed!",
+					      quoteStr);
+		return NULL;
+	}
+
+	if (strstr (dsConfig->metaTableKeyColName, quoteStr))
+	{
+		ELEKTRA_SET_INTERFACE_ERRORF (errorKey,
+					      "The specified name of the key column in the metatable contained the quote string "
+					      "'%s', this is not allowed!",
+					      quoteStr);
+		return NULL;
+	}
+
+	if (strstr (dsConfig->metaTableMetaKeyColName, quoteStr))
+	{
+		ELEKTRA_SET_INTERFACE_ERRORF (errorKey,
+					      "The specified name of the metakey column in the metatable contained the quote "
+					      "string '%s', this is not allowed!",
+					      quoteStr);
+		return NULL;
+	}
+
+	/* Example: DELETE FROM metaTable WHERE keyName=myKey AND metaKeyName IN (mk1, mk2, mk3) */
```

Comment:
Just as a head's up: the `IN` clause can't be of arbitrary length, there is a limit (which often isn't even clearly documented) on how many elements you can pass in it. I've had much fun debugging and fixing those things in my day job. But I think we probably won't reach it except for huge benchmarks.